### PR TITLE
Super Budget app

### PR DESCRIPTION
**Pull request by sentayhu19**
**In this PR**

- [x] Splash screen
  - A simple page with the name of Super budget , and links to the sign up and log in pages.

- [x]  Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- [x] Home page (catagories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- [x] Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- [x] "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user click (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- [x] "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new category, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

#### Testing requirements
- [x] Unit and integration tests for all the most important components of RoR application.
- [x] Deployed here -[Live](https://super-budget.herokuapp.com)
- [x] [Video Presentation](https://www.loom.com/share/fa2829ce46d64e49a838f5e0832a34c6)

mobile001
mobile002
>

### CODE REVIEW

File: app/controllers/categories_controller.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+  # GET /categories/1/edit
+  def edit; end
+
+  # POST /categories or /categories.json
+  def create
+    @category = Category.new(category_params)
+    respond_to do |format|
+      if @category.save
+        format.html { redirect_to categories_path, notice: 'Category was successfully created.' }
```
Comment (aimemalaika) [inline app/controllers/categories_controller.rb]: - [x] Kindly implement the show route to allow to view a single category, it details and transactions associated to it

**Review by aimemalaika — CHANGES_REQUESTED** [review]
Hi @sentayhu19, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

the-simpsons-homer-simpson


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/categories_controller.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  # GET /categories or /categories.json
+  def index
+    @categories = Category.where(user: current_user)
+  end
+
+  # GET /categories/1 or /categories/1.json
+  # def show
+  #   @transactions1s = @category.transactions1
+  # end
+
```
Comment (aimemalaika) [inline app/controllers/categories_controller.rb]: - [x] Kindly remove commented code on your online repository

File: app/controllers/categories_controller.rb (hunk 1 above)
Comment (rloterh) [reply app/controllers/categories_controller.rb]: This issue is well resolved! ✔️

File: app/controllers/categories_controller.rb (hunk 2 above)
Comment (rloterh) [reply app/controllers/categories_controller.rb]: This issue is well resolved! ✔️

File: README.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+- GitHub: [@sentayhu19](https://github.com/sentayhu19)
+- LinkedIn: [sentayhu-berhanu](https://www.linkedin.com/in/sentayhu-berhanu-6376579a/)
+
+## 🤝 Contributing
+
+Contributions, issues, and feature requests are welcome!
+
+Feel free to check the [issues page](https://github.com/sentayhu19/Budget-app/issues).
+
```
Comment (rloterh) [inline README.md:78]: - [x] OPTIONAL
It's a nice and smart thing to give appropriate credit to **[Gregoire Vella](https://www.behance.net/gregoirevella)** on Behance, the author of [the original design](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as required by the [Creative Commons License](https://creativecommons.org/licenses/). Kindly ensure to add this under the **Acknowledgements** section of your ReadMe.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Sentayhu,  

Your project is complete! There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉🌟 

## Highlights 👍
- App is running as intended

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

_Kindly consider my comments marked **[OPTIONAL]** under this review_
<br>
keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**